### DISCUSSION TITLE
Discovery Version not available

### DISCUSSION THREAD

**Discussion by aglowinski-danfoss** [ansys/pyansys-geometry#1491] [Q&A] [answered]
Dear developers,
I am running a script on 2 different PCs with the same version of PyAnsys Geometry (0.6.5) and the same version of Ansys installed (24R2). On one PC I am not able to launch Discovery with the command: 
modeler = launch_modeler_with_discovery(product_version=242)
I get the error:
SystemError: The requested Ansys product's version 242 is not available, please specify a different version.
Am I missing somethin here?
Best regards,
Alex

**Comment by RobPasMue** [comment]
Hi @aglowinski-danfoss - thanks for opening this discussion. Let's try to figure it out. Can you share with us (from your failing environment):

1. What version of ``ansys-tools-path`` you are using? (``pip list`` and search for this package)
2. Can you also run the following code and share the output?
```py
from ansys.tools.path import get_available_ansys_installations, get_latest_ansys_installation

installs = get_available_ansys_installations()
print(installs)

latest = get_latest_ansys_installation()
print(latest)

print("24R2 is not available") if 242 not in installs else print("24R2 is available")
```

**Reply by aglowinski-danfoss** [reply]
Hi @RobPasMue ,
1. Ansys Tools Path is 0.3.2 for both PCs (they got the same VENV)
2.
{232: 'C:\\Program Files\\ANSYS Inc\\v232'}
(232, 'C:\\Program Files\\ANSYS Inc\\v232')
24R2 is not available

The PC where it worked gives me:
{222: 'C:\\Program Files\\ANSYS Inc\\v222', 231: 'C:\\Program Files\\ANSYS Inc\\v231', 232: 'C:\\Program Files\\ANSYS Inc\\v232', 241: 'C:\\Program Files\\ANSYS Inc\\v241', 242: 'C:\\Program Files\\ANSYS Inc\\v242', -242: 'C:\\Program Files\\ANSYS Inc\\ANSYS Optics\\v242'}
(242, 'C:\\Program Files\\ANSYS Inc\\v242')

**Reply by RobPasMue** [reply]
As you can see, on your failing machine, 242 is not being recognized. Let's do some checks:

1. Are we sure that ``C:\Program Files\ANSYS Inc\v242`` exists?
2. Do you have the environment variables ``ANSYS242_DIR`` and ``AWP_ROOT242`` defined?

**Reply by aglowinski-danfoss** [reply]
1. yes
2.  ANSYS242_DIR is defined under: C:\Program Files\ANSYS Inc\v242\ANSYS
AWP_ROOT242 is defined under C:\Program Files\ANSYS Inc\v242
I also did some restarts, because I also had the feeling, it has something to do with the environment variables.

**Reply by RobPasMue** [reply]
Did it work after the restarts?

**Reply by aglowinski-danfoss** [reply]
No, unfortunately not. There were also some old ansys versions in the environment variables, that I removed. I still only get the 232 listed.

**Reply by RobPasMue** [reply]
Hmm ``ansys-tools-path==0.3.2`` is pretty old in fact... and that got me thinking

https://github.com/ansys/ansys-tools-path/blob/9650ee1f8bb1b8b4af03f2798149ba0ce5a043e0/src/ansys/tools/path/path.py#L24-L40

This is the code in 0.3.2 and as you can see, version 24R2 was not included yet! So...

Can you update ansys-tools-path to the latest version?

```bash
pip install --upgrade --force-reinstall ansys-tools-path
```

This should fix it!

**Reply by RobPasMue** [reply]
Also, I'd be surprised that your working machine also has ```ansys-tools-path==0.3.2``` and works fine... please verify that the versions are the same.

**Comment by RobPasMue** [comment, marked as answer]
For summary purposes on future reads...

After trying to determine information relative to:

> * What version of ansys-tools-path you are using? (pip list and search for this package)
> * Can you also run the following code and share the output?
> 
> ```py
> from ansys.tools.path import get_available_ansys_installations, get_latest_ansys_installation
> 
> installs = get_available_ansys_installations()
> print(installs)
> 
> latest = get_latest_ansys_installation()
> print(latest)
> 
> print("24R2 is not available") if 242 not in installs else print("24R2 is available")
> ```

We realized that the ``ansys-tools-path`` version being used was too old (0.3.2) and did not have "knowledge" of the Ansys 24R2 release. A version upgrade of ``ansys-tools-path`` made the original script work:

```bash
pip install --upgrade --force-reinstall ansys-tools-path
```

If you agree with this resolution @aglowinski-danfoss please mark this response as an answer.
